src/vm/object.cpp

```diff
@@ -1969,54 +1969,120 @@ StackTraceElement & StackTraceArray::operator[](size_t index)
 }
 
 #if !defined(DACCESS_COMPILE)
-// Define the lock used to access stacktrace from an exception object
+// Define the lock used to access stacktrace from a global exception type
 SpinLock g_StackTraceArrayLock;
 
-void ExceptionObject::SetStackTrace(StackTraceArray const & stackTrace, PTRARRAYREF dynamicMethodArray)
+void ExceptionObject::SetStackTrace(StackTraceArray const * stackTrace, PTRARRAYREF dynamicMethodArray)
 {        
     CONTRACTL
     {
-        GC_NOTRIGGER;
-        NOTHROW;
+        THROWS;
+        GC_TRIGGERS;
         MODE_COOPERATIVE;
     }
     CONTRACTL_END;
+    
+    OBJECTREF trace = NULL;
+    if (stackTrace != NULL)
```

Comment:
Nit: You can change the argument to `I1ARRAYREF`, move the `Get()` to the caller, and avoid this if-check.

It would also make the method signature look better - both arguments would be of the same kind.

Reply:
Changed